api/Controllers/MiniUrl.cs

```diff
@@ -14,25 +11,20 @@ public class MiniUrlController : ControllerBase
 {
     
     private readonly ILogger<MiniUrlController> _logger;
-    private readonly IConfiguration _config;
-    private readonly string _domain;
-    private readonly IDatabase _redis0; // shortened url store
-    private readonly IDatabase _redis1; // rate limiting management store
+    private readonly string? _domain;
+    private readonly RedisActions _redis;
     
     
     public MiniUrlController(IConfiguration configuration, ILogger<MiniUrlController> logger, IConnectionMultiplexer multiplexer)
     {
-
-        _config = configuration;
         _logger = logger;
-        _domain = _config["Domain"];
-        _redis0 = multiplexer.GetDatabase(0);
-        _redis1 = multiplexer.GetDatabase(1);
+        _domain = configuration["Domain"];
+        _redis = new RedisActions(multiplexer.GetDatabase(0), multiplexer.GetDatabase(1));
     }
 
     
     [HttpPost(Name = "Minify")]
-    public ActionResult<MinifyResponse> Post([FromBody] MinifyRequest body)
+    public async Task<MinifyResponse> Post([FromBody] MinifyRequest body)
```

Comment:
Task<ActionResult<MinifyResponse>> is better, since it allows you to return proper HTTP response in case of issue.

Reply:
done